tests/test_dgidb.py

```diff
@@ -1,94 +1,209 @@
-    assert len(results) == 0
-
-    # Use pandas default is correctly set to 'true'
-    query = "braf"
-    results = get_interactions(query)
-    assert type(results) == type(pd.DataFrame())
-
-    # Use pandas can be toggled to 'false' and returns a dictionary response object
-    query = "braf"
-    results = get_interactions(query, use_pandas=False)
-    assert type(results) != type(pd.DataFrame())
-    assert type(results) == type(dict())
-
-    # Gene search types work correctly
-    query = "braf"
-    results = get_interactions(query, search="genes")
-    assert results.columns[0] == "gene"
-
-    # Gene search is not grabbing drugs
-    query = "imatinib"
-    results = get_interactions(query, search="genes")
-    assert len(results) == 0
-
-    # Drug search types work correctly
-    query = "imatinib"
-    results = get_interactions(query, search="drugs")
-    assert type(results) == type(pd.DataFrame())
-    assert results.columns[0] == "drug"
-
-    # Drug search is not grabbing genes
-    query = "XPO1"
-    results = get_interactions(query, search="drugs")
-    assert len(results) == 0
-
-
-def test_get_categories():
-    """Test that categories works correctly"""
-    # Categories search builds a dataframe (with use_pandas default set to 'true')
-    query = "braf"
-    results = get_categories(query)
-    assert type(results) == type(pd.DataFrame())
-
-    # Categories does not return drugs data
-    query = "imatinib"
-    results = get_categories(query)
-    assert len(results) == 0
-
-    # Use pandas can be toggled to 'false' and returns a dictionary response object
-    query = "imatinib"
-    results = get_categories(query, use_pandas=False)
-    assert type(results) != type(pd.DataFrame())
-    assert type(results) == type(dict())
-
-
-def test_get_drugs():
-    """Test that drug profile works correctly"""
-    # Drug search builds a dataframe (with use_pandas default set to 'true')
-    query = "imatinib"
-    results = get_drug(query)
-    assert type(results) == type(pd.DataFrame())
-
-    # Drug search does not return gene data
-    query = "XPO1"
-    results = get_drug(query)
-    assert (len(results)) == 0
-
-    # Use pandas can be toggled to 'false' and returns dictionary response object
-    query = "imatinib"
-    results = get_drug(query, use_pandas=False)
-    assert type(results) != type(pd.DataFrame())
-    assert type(results) == type(dict())
-
-
+import requests_mock
+
+from dgipy.dgidb import (
+    get_categories,
+    get_drug,
+    get_drug_applications,
+    get_gene,
+    get_gene_list,
+    get_interactions,
+    get_source,
+)
+
+
+def test_get_drugs(fixtures_dir: Path, set_up_graphql_mock: Callable):
+    with requests_mock.Mocker() as m, (
+        fixtures_dir / "get_drug_api_response.json"
+    ).open() as json_response, (
+        fixtures_dir / "get_drug_filtered_api_response.json"
+    ).open() as filtered_json_response:
+        set_up_graphql_mock(m, json_response)
```

Comment:
note the general pattern here -- requests_mock gives us a context manager within which we can define legal HTTP requests and inject our own response values